src/Payments/Contracts/PlaidActionsTrait.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Kanvas\Packages\Payments\Contracts;
+
+use function Baka\envValue;
+use Canvas\Validation as CanvasValidation;
+use Phalcon\Http\Response;
+use Phalcon\Validation\Validator\PresenceOf;
+use Stripe\Charge;
+use Stripe\Stripe;
+use TomorrowIdeas\Plaid\PlaidRequestException;
+
+trait PlaidActionsTrait
+{
+    /**
+     * plaidAuth function
+     * get plaid account form access token and return ach numbers account.
+     *
+     * @return Response
+     */
+    public function plaidAuth() : Response
+    {
+        $validation = new CanvasValidation();
+        $validation->add('public_token', new PresenceOf(['message' => _('The access_token is required.')]));
+        $request = $this->request->getPostData();
+        $validation->validate($request);
+        $response = $this->plaid->exchangeToken($request['public_token']);
+        $accessToken = $response->access_token;
+        $auth = $this->plaid->getAuth($accessToken);
+        if (!$auth->numbers || !$auth->numbers->ach) {
+            throw new Exception('We have a problem on palid account');
+        }
+        return $this->response($auth->numbers->ach);
+    }
+
+    /**
+     * plaidTransaction function
+     * make a plaid charge through stripe.
+     *
+     * @return Response
+     */
+    public function plaidTransaction() : Response
+    {
+        $request = $this->request->getPostData();
+        $validation = new CanvasValidation();
+        $validation->add('public_token', new PresenceOf(['message' => _('The access_token is required.')]));
+        $validation->add('account_id', new PresenceOf(['message' => _('The access_token is required.')]));
+        $validation->add('amount', new PresenceOf(['message' => _('The access_token is required.')]));
+        $validation->validate($request);
+
+        $charge = $this->makeTransaction($request);
```

Comment:
@FredPeal we dont need this function no? since this is call directly 1 time in the controller